toolchain/sem_ir/formatter.h

```diff
@@ -119,9 +119,11 @@ class Formatter {
   auto IndentLabel() -> void;
 
   // Formats a top-level scope, and any of the instructions in that scope that
-  // are used.
-  auto FormatScopeIfUsed(InstNamer::ScopeId scope_id,
-                         llvm::ArrayRef<InstId> block) -> void;
+  // are used. Instructions may optionally use `TentativeOutputScope` (see type
+  // for details).
```

Comment:
This phrasing makes it unclear whether this is giving permission to the caller or the callee, and how it relates to `is_tentative`.

```suggestion
  // are used. If `is_tentative` is true, the instructions will be formatted using 
  // `TentativeOutputScope` (see type for details).
```

I think it would be better to document this without relying so heavily on implementation details like `TentativeOutputScope`, but I don't understand `TentativeOutputScope` well enough to suggest how to phrase that.

Reply:
> This phrasing makes it unclear whether this is giving permission to the caller or the callee, and how it relates to is_tentative.

How about this: renamed to `use_tentative_output_scopes` for explicitness, dropped the comment since the correlation between `use_tentative_output_scopes` and `TentativeOutputScope` should be more implicit.

> I think it would be better to document this without relying so heavily on implementation details like `TentativeOutputScope`, but I don't understand `TentativeOutputScope` well enough to suggest how to phrase that.

This is getting to complex `Formatter` behavior, so while maybe that could be more clearly documented, I think adding more detail here would end up duplicative, which I want to avoid. And, I don't think this is the right place to document that behavior.

Reply:
This comment and my other comment above are getting at the same underlying issue: I spent maybe half an hour trying to understand the changes to this function well enough to e.g. be able to decide when to pass `true` vs `false`, including looking quite closely at the comments and implementation of `TentativeOutputScope`, and I'm still confused. Some of that is attributable to my unfamiliarity with this part of the formatter, but I suspect that a few comments in key places could have reduced that time substantially.

Reply:
Maybe good to bring up on toolchain? But, does it need to block this PR?

Reply:
To be sure -- my perspective is I'm pretty much just moving the `ShouldFormatInst` call from inside `FormatInst` to instead put it at the caller. And, if `ShouldFormatInst` is false, there's really no reason to put it in a `TentativeOutputScope`. I believe the tests show the changed code is pretty much as expected.

Reply:
> Maybe good to bring up on #toolchain?

Sure, will do.

> But, does it need to block this PR?

In principle maybe not, but in practice I don't feel like I can review the code if I don't understand it.

Reply:
We'd discussed a TODO, but given discussion on #toolchain, maybe it's better to just do the condition swap in this PR and see how it works out?

Reply:
That change looks great, but I would've expected it to have some effect on the test outputs (e.g. starting to print more constants and imports). Am I misunderstanding something?

Reply:
This sample has no constants or imports. See #5455 for a more complex example.

Reply:
n/m, fixed. But still, note #5455 is the more complex case.